### ISSUE TITLE
Test fails if port 1234 is in use on the test host

### ISSUE THREAD

**Issue by matthewbentley** [Yelp/paasta#1276]
I had znc set up to listen on port 1234, causing `test_run_docker_container_with_user_specified_port` to fail with `The chosen port is already in use!`.

I don't know if this should be fixed (by looking for an unused port first) or documented somewhere.

**Comment by Rob-Johnson** [comment]
yeah, the test can be made more resilient by looking for a free port, rather than hardcoding one.
